resources/splunk_install.rb

```diff
@@ -0,0 +1,223 @@
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :version, String, required: true
+  property :build, String, required: true
+  property :base_url, String, default: 'https://download.splunk.com/products'
+  property :accept_license, [TrueClass], required: true
+
+  default_action :install
+
+  def after_created
+    node.run_state['splunk_ingredient'] ||= { 'installations' => {} }
+
+    case name.downcase
+    when 'splunk' then package :splunk
+    when 'universal_forwarder' then package :universal_forwarder
+    else raise 'Package must be specified (:splunk or :universal_forwarder)'
+    end unless property_is_set? :package
+  end
+
+  ### Inherited Methods
+
+  # Must be overridden by a platform provider
+  def kernel_string
+    raise 'No implementation for current platform'
+  end
+
+  def package_name
+    case package
+    when :splunk then 'splunk'
+    when :universal_forwarder then 'splunkforwarder'
+    end
+  end
+
+  def install_dir
+    @install_dir ||= default_install_dirs[package][node['os'].to_sym]
+    raise "Unsupported Combination: #{package} + #{node['os']}" unless @install_dir
+    @install_dir
+  end
+
+  def package_url
+    @package_url ||=
+      case package
+      when :splunk then "#{base_url}/splunk/releases/#{version}/#{node['os']}/splunk"
+      when :universal_forwarder then "#{base_url}/universalforwarder/releases/#{version}/#{node['os']}/splunkforwarder"
+      end + "-#{version}-#{build}-#{kernel_string}"
+  end
+
+  def package_path
+    @package_path ||= Pathname.new(Chef::Config['file_cache_path']) + filename_from_url(package_url)
+  end
+
+  ### Inherited Actions
+
+  load_current_value do |desired|
+    package desired.package
+    version_file = version_pathname
+    current_value_does_not_exist! unless version_file.exist?
+
+    version_data = version_file.read
+    installed_version, installed_build = version_data.match(/VERSION=(\d(?:\.\d)+).+BUILD=([\w]+)/m).captures
+    version installed_version
+    build installed_build
+  end
+
+  action :install do
+    install_state = node.run_state['splunk_ingredient']
+    raise "Install at #{install_dir} already exists!" if install_state['installations'][install_dir]
+    install_state['current_installation'] = install_state['installations'][install_dir] = {
+      name: name,
+      package: package,
+      version: version,
+      build: build,
+      x64: x64_support
+    }
+
+    converge_if_changed :version, :build do
+      remote_file package_path.to_s do
+        source package_url
+        show_progress true if defined? show_progress # Chef 12.9 feature
+        notifies :delete, "remote_file[#{package_path}]", :delayed
+      end
+    end
+  end
+
+  action :uninstall do
+    install_state = node.run_state['splunk_ingredient']
+    install_state.delete('current_installation')
```

Comment:
That's a curious question, because it's easier and more sanitary in my mind to remove the current_installation as that entry in run_state is meant for chaining future resources and performing actions immediately after an install. It would be very unusual to do so after an uninstall, and I don't think we need that logic right now. It's not an impossible scenario, though, either. Just maybe one that has some smelly code.

Furthermore, I intend to follow the pattern of setting current_installation in every resource that specifies an installation, so with uninstall not having an installation anymore to set this to, it makes more sense that there is no current_installation.